### ISSUE TITLE
IoU values coming out as negative or > 1

### ISSUE THREAD

**Issue by ryx2** [qubvel/segmentation_models#71]
How can this happen? Loss is also negative, model always converges to predicting all class 0 

`genargs = dict(
    brightness_range=[0.5, 2],
    rotation_range=180,
    zoom_range=[0.3,2],
    horizontal_flip=True,
    vertical_flip=True
)

xgen = ImageDataGenerator(**genargs)
ygen = ImageDataGenerator(**genargs)

# compute quantities required for featurewise normalization
# (std, mean, and principal components if ZCA whitening is applied)
seed = 1
xgen.fit(x_train, augment=True, seed = seed)
ygen.fit(y_train, augment=True, seed = seed)

epochs = 1000
saves = 10

# fits the model on batches with real-time data augmentation:

###seg model

BACKBONE = 'inceptionv3'
x_train=sm.backbones.get_preprocessing(BACKBONE)(x_train)
#need to get them to 528 x 672

x_train = np.pad(x_train, ((0,0), (0,16),(0,32),(0,0)), 'mean')
x_train = np.concatenate((x_train,x_train,x_train), axis = 3)
# add the padding as background class
y_train = np.stack([np.pad(y_train[:,:,:,0], ((0,0), (0,16),(0,32)), 'constant', constant_values=1),
np.pad(y_train[:,:,:,1], ((0,0), (0,16),(0,32)), 'constant', constant_values=0),
np.pad(y_train[:,:,:,2], ((0,0), (0,16),(0,32)), 'constant', constant_values=0)], axis = 3)

model = sm.PSPNet(BACKBONE, input_shape=(528,672,3), classes=3, encoder_weights=None, psp_use_batchnorm=False)
model.compile(keras.optimizers.Adam(), #lr=1e-4, epsilon=1e-8, decay=1e-6
              loss=sm.losses.jaccard_loss, metrics=[sm.metrics.f_score])

batch_size=4
initial_epoch = 0

if train:

    x_train = y_train

    xgen = xgen.flow(x_train, batch_size=batch_size, seed=seed)
    # class mygen:
    #     def __next__(self):
    #         yield sm.backbones.get_preprocessing(BACKBONE)(xgen.next())
    ygen = ygen.flow(y_train, batch_size=batch_size, seed=seed)
    train_generator = zip(xgen, ygen)

    subset = round(epochs / saves)
    for current_epoch in range(subset):
        model.fit_generator(train_generator, #class_weight={0: 1, 1: 5, 2: 1},
                        steps_per_epoch=len(x_train) / batch_size, epochs=saves, verbose = 1, initial_epoch=initial_epoch)`

**Comment by ryx2** [comment]
I'm setting 
x_train = y_train
as a test, seeing if the model can just predict the input as its output.

**Comment by qubvel** [comment]
Input images should be in range (0, 255)
Ground truth masks - binary images {0, 1}
